### ISSUE TITLE
add GPU execution mode option to Sirepo

### ISSUE THREAD

**Issue by moellep** [radiasoft/sirepo#6723]
Add a new execution mode "GPU" for running activait sims. The current "5 cores" option takes too long (hours) for image ML processing and could be done much faster on a GPU.

This feature would require new infrastructure for resource management and allocation of a limited number of GPUs available to Sirepo.

**Comment by e-carlin** [comment]
Notes from discussion with Rob:
- Run the service on the gpu box with less RAM. Connect it to sirepo alpha
- Users must be premium customers to get access to gpus
- Users will select whether they want to run on gpu or not
- "parallel" jobs will be the only ones that can be run on gpus
- We need to add the concept of heterogenous compute nodes to sirepo. Currently we assume all nodes have the same resources. It might be worthwhile to add understanding of cpu cores and memory while doing this. Again, we currently assume all nodes have the same number of cores and same ram which isn't true. At the very least we'll need to know some nodes have gpus and others dont
- When a user requests a gpu compute we'll need to stop their parallel agent running on a non-gpu node, start a parallel agent on a gpu node, and send the job. Sequential agents don't need to be stopped. It may be time to implement the concept of parked agents while doing this work.
- GPU computes will probably become a new job_run_mode type. It would be good to fix confusion between isParallel and job_run_mode while doing this work. Currently a user can select serial execution but a job isParallel (has animation in the name) which can lead to jobs trying to run in parallel on an agent with only one core.
